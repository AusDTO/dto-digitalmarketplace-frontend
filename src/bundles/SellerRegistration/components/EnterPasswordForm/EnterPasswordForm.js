import React from 'react';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';

import { min } from '../../../../validators';

import Layout        from '../../../../shared/Layout';
import BaseForm      from '../../../../shared/form/BaseForm';
import SubmitForm    from '../../../../shared/form/SubmitForm';
import ErrorBox      from '../../../../shared/form/ErrorBox';
import Textfield     from '../../../../shared/form/Textfield';
import formProps     from '../../../../shared/reduxModules/formPropsSelector';

class EnterPasswordForm extends BaseForm {

  static propTypes = {
    action: React.PropTypes.string,
    csrf_token: React.PropTypes.string,
    form: React.PropTypes.object.isRequired
  }

  render() {
    const { action, csrf_token, model, form, onSubmit, enterPasswordForm } = this.props;
    return (
      <Layout>
        <header>
          <h1>Create a password</h1>
          <p>To finish creating your account you need to create a password.</p>
        </header>
        <article role="main">
          <ErrorBox focusOnMount={true} model={model}/>
          <Form model={model}
            action={action}
            method="post"
            id="enterpassword"
            component={SubmitForm}
            valid={form.valid}
            onCustomSubmit={onSubmit}
          >
            {csrf_token && (
              <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token} />
            )}

            <div className="field">
              <div className="question-heading">Email</div>
              <p>{enterPasswordForm.email}</p>
            </div>

            <Textfield
              model={`${model}.password`}
              name="password"
              id="password"
              htmlFor="password"
              label="Password"
              description="At least 10 characters"
              validators={{ min : min(10) }}
              messages={{
                min: 'Your password must be at least 10 characters',
              }}
              type="password"
            />

            <div>
              All activity on the Digital Marketplace is subject to our <a href="/terms-of-use">Terms of Use</a>.
            </div>

            <input type="submit" value='Join the Marketplace' role="button" />
          </Form>
        </article>
      </Layout>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    ...formProps(state, 'enterPasswordForm')
  }
}

export {
  Textfield,
  mapStateToProps,
  EnterPasswordForm as Form
}

export default connect(mapStateToProps)(EnterPasswordForm);
