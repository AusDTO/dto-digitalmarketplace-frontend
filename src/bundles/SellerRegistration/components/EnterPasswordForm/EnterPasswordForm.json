{
  "enterPasswordForm": {
    "representative": "Joe Bloggs",
    "name": "My Company",
    "abn": "1234567",
    "phone": "55555555",
    "email": "test@company.com"
  },
  "form_options": {
  }
}
