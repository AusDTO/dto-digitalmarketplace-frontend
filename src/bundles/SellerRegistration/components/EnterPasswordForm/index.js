import EnterPasswordForm from './EnterPasswordForm'
export default EnterPasswordForm
