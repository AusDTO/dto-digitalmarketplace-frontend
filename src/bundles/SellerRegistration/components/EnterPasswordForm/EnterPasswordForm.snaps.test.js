// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import EnterPasswordForm, { Textfield } from './EnterPasswordForm'
import sampleState from './EnterPasswordForm.json'
import createStore from '../../redux/create'

test('EnterPasswordForm renders', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }))
  const component = renderer.create(
  	<Provider store={store}>
    	<EnterPasswordForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('EnterPasswordForm renders with form_options', () => {
  const form_options = {
    csrf_token: 'sometoken',
    action: '/foo/bar'
  }
  let store = createStore(Object.assign({}, { _serverContext: {}, form_options }))
  const component = renderer.create(
    <Provider store={store}>
      <EnterPasswordForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('EnterPasswordForm renders with populated fields', () => {
  let store = createStore(Object.assign({}, { _serverContext: {} }, sampleState))
  const component = renderer.create(
    <Provider store={store}>
      <EnterPasswordForm />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

