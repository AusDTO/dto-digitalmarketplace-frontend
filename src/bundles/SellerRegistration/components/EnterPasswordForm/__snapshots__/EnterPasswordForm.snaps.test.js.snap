exports[`test EnterPasswordForm renders 1`] = `
<div
  className="layout">
  <header>
    <h1>
      Create a password
    </h1>
    <p>
      To finish creating your account you need to create a password.
    </p>
  </header>
  <article
    role="main">
    <form
      action={undefined}
      id="enterpassword"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}>
      <div
        className="field">
        <div
          className="question-heading">
          Email
        </div>
        <p />
      </div>
      <div
        className="field">
        <label
          className="question-heading"
          htmlFor="password">
          Password
        </label>
        <p
          className="hint"
          id="password-hint">
          At least 10 characters
        </p>
        <input
          aria-describedby="password-hint"
          className=""
          disabled={false}
          id="password"
          maxLength={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          type="text"
          value="" />
      </div>
      <div>
        All activity on the Digital Marketplace is subject to our 
        <a
          href="/terms-of-use">
          Terms of Use
        </a>
        .
      </div>
      <input
        role="button"
        type="submit"
        value="Join the Marketplace" />
    </form>
  </article>
</div>
`;

exports[`test EnterPasswordForm renders with form_options 1`] = `
<div
  className="layout">
  <header>
    <h1>
      Create a password
    </h1>
    <p>
      To finish creating your account you need to create a password.
    </p>
  </header>
  <article
    role="main">
    <form
      action="/foo/bar"
      id="enterpassword"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}>
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="sometoken" />
      <div
        className="field">
        <div
          className="question-heading">
          Email
        </div>
        <p />
      </div>
      <div
        className="field">
        <label
          className="question-heading"
          htmlFor="password">
          Password
        </label>
        <p
          className="hint"
          id="password-hint">
          At least 10 characters
        </p>
        <input
          aria-describedby="password-hint"
          className=""
          disabled={false}
          id="password"
          maxLength={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          type="text"
          value="" />
      </div>
      <div>
        All activity on the Digital Marketplace is subject to our 
        <a
          href="/terms-of-use">
          Terms of Use
        </a>
        .
      </div>
      <input
        role="button"
        type="submit"
        value="Join the Marketplace" />
    </form>
  </article>
</div>
`;

exports[`test EnterPasswordForm renders with populated fields 1`] = `
<div
  className="layout">
  <header>
    <h1>
      Create a password
    </h1>
    <p>
      To finish creating your account you need to create a password.
    </p>
  </header>
  <article
    role="main">
    <form
      action={undefined}
      id="enterpassword"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}>
      <div
        className="field">
        <div
          className="question-heading">
          Email
        </div>
        <p>
          test@company.com
        </p>
      </div>
      <div
        className="field">
        <label
          className="question-heading"
          htmlFor="password">
          Password
        </label>
        <p
          className="hint"
          id="password-hint">
          At least 10 characters
        </p>
        <input
          aria-describedby="password-hint"
          className=""
          disabled={false}
          id="password"
          maxLength={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          pattern={undefined}
          type="text"
          value="" />
      </div>
      <div>
        All activity on the Digital Marketplace is subject to our 
        <a
          href="/terms-of-use">
          Terms of Use
        </a>
        .
      </div>
      <input
        role="button"
        type="submit"
        value="Join the Marketplace" />
    </form>
  </article>
</div>
`;
